src/webargs/core.py

```diff
@@ -198,19 +201,23 @@ def _get_loader(self, location: str) -> typing.Callable:
 
     def _load_location_data(
         self, *, schema: ma.Schema, req: Request, location: str
-    ) -> typing.Mapping:
+    ) -> typing.Any:
         """Return a dictionary-like object for the location on the given request.
 
         Needs to have the schema in hand in order to correctly handle loading
         lists from multidict objects and `many=True` schemas.
         """
         loader_func = self._get_loader(location)
-        data = loader_func(req, schema)
-        # when the desired location is empty (no data), provide an empty
-        # dict as the default so that optional arguments in a location
-        # (e.g. optional JSON body) work smoothly
-        if data is missing:
-            data = {}
```

Comment:
What's the rationale for removing this?

Reply:
It was moved into `_process_location_data`. Throughout the changes, I tried to keep the async vs sync copies of methods as short as possible. If they had a common piece of logic -- like this -- it moved to a new location if possible/feasible.

Reply:
Oh, right. I should have `Ctrl+F`ed that